### PR TITLE
add 'variable' and 'group' kwargs to 'rioxarray.open_rasterio()' to allow filtering of subdatasets

### PR DESCRIPTION
**Pull request by snowman2**
- [x] Tests added
 - [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

### CODE REVIEW

File: rioxarray/_io.py [hunk 1]
```diff
@@ -194,14 +195,49 @@ def parsevec(s):
...
+ 
+    Returns
+    -------
+    re.SRE_Pattern: output of re.compile()
+    """
+    variable_query = r"\w+"
+    if variable_names is not None:
+        if not isinstance(variable_names, (tuple, list)):
+            variable_names = [variable_names]
+        variable_query = rf"(?:{'|'.join(variable_names)})"
```
Comment (justingruca) [inline rioxarray/_io.py]: Are there any restrictions on the syntax of `variable_names`? Might want to run them through [`re.escape`](https://docs.python.org/3/library/re.html#re.escape).

File: rioxarray/_io.py [hunk 2]
```diff
@@ -194,14 +195,49 @@ def parsevec(s):
...
+    if group_names is not None:
+        if not isinstance(group_names, (tuple, list)):
+            group_names = [group_names]
+        group_query = rf"(?:{'|'.join(group_names)})"
```
Comment (justingruca) [inline rioxarray/_io.py]: Same with `group_names`.

File: rioxarray/_io.py [hunk 3]
```diff
@@ -194,14 +195,49 @@ def parsevec(s):
     return parsed_tags
 
 
+def build_subdataset_filter(group_names, variable_names):
+    """
+    Examples:
+'HDF4_EOS:EOS_GRID:"./modis/MOD09GQ.A2017290.h11v04.006.NRT.hdf":MODIS_Grid_2D:sur_refl_b01_1'
```
Comment (justingruca) [inline rioxarray/_io.py]: indent?

Comment (snowman2) [reply rioxarray/_io.py]: Fixed that one.

File: rioxarray/_io.py (hunk 1 above)
Comment (snowman2) [reply rioxarray/_io.py]: Good suggestion. Added it in. Thanks!

**Pull request merged**